### ISSUE TITLE
Error finding 'label'

### ISSUE THREAD

**Issue by Henkkas52** [reneklootwijk/node-dsmr#34]
Most of the time dsmr-parser (2.04) is working well, but every now and then I get an error in debug window of node-red. 
Undefined "label" ???

![afbeelding](https://github.com/reneklootwijk/node-dsmr/assets/80907008/3b88b692-d86a-42cf-b298-ec92e8dc28b8)

This is a normal P1 message:
 
`/Ene5\T211 ESMR 5.0

1-3:0.2.8(50)
0-0:1.0.0(230902152902S)
0-0:96.1.1(4530303632303030303130313639363232)
1-0:1.8.1(000352.869*kWh)
1-0:1.8.2(000124.566*kWh)
1-0:2.8.1(000987.978*kWh)
1-0:2.8.2(001760.544*kWh)
0-0:96.14.0(0001)
1-0:1.7.0(00.000*kW)
1-0:2.7.0(01.319*kW)
0-0:96.7.21(00008)
0-0:96.7.9(00004)
1-0:99.97.0(1)(0-0:96.7.19)(220816003006S)(0000000380*s)
1-0:32.32.0(00001)
1-0:52.32.0(00001)
1-0:72.32.0(00002)
1-0:32.36.0(00000)
1-0:52.36.0(00000)
1-0:72.36.0(00000)
0-0:96.13.0()
1-0:32.7.0(226.0*V)
1-0:52.7.0(226.0*V)
1-0:72.7.0(227.0*V)
1-0:31.7.0(001*A)
1-0:51.7.0(001*A)
1-0:71.7.0(003*A)
1-0:21.7.0(00.000*kW)
1-0:41.7.0(00.000*kW)
1-0:61.7.0(00.000*kW)
1-0:22.7.0(00.218*kW)
1-0:42.7.0(00.322*kW)
1-0:62.7.0(00.778*kW)
0-1:24.1.0(003)
0-1:96.1.0(4730303733303034313539343039353232)
0-1:24.2.1(230902152500S)(00013.636*m3)
!4774
`

**Comment by reneklootwijk** [comment]
Is that all? I need some more information to determine where in the code this error occurs.

**Comment by Henkkas52** [comment]
My guess was, the error occurs within the dsmr-parser itself.
But I will try to describe how my code looks like.

I made a function node, with following code:

`// if not "msg.payload" then stop
if (msg.hasOwnProperty("payload") == false) {
    return;
}

// If disabled then stop
let enabled = flow.get('enabled');
if (!enabled) {
    return;
}


const parser = global.get('dsmrParser');

let telegram = msg.payload;

let begin = telegram.includes("/");
let end = telegram.includes("!");

if (!begin || !end) {
    // Filter incomplete messages 
    msg = null;
}
else {
    // Complete message split and filter empty array elements
    let arr_new = [];
    let arr = telegram.split(/[\n\r]/g);
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] != "") {
            arr_new.push(arr[i]);
        }
    }
    msg = { topic: "telegram", telegram: msg.payload, parser: parser(arr_new), port: msg.port}
}


return msg;
`

The input data for this function comes from a serial in node.
Then for this occasion I defined an error node on my dsmr function in order to try to get more info,
The output looks like this:


![afbeelding](https://github.com/reneklootwijk/node-dsmr/assets/80907008/e7dddb77-f696-4ced-a4e9-a092700c5bba)

**Comment by reneklootwijk** [comment]
I do not have any idea what causes this issue, but I do not know anything about node-red, what version of my module it uses and how it uses my module. Do you know which version is being used?

**Comment by Henkkas52** [comment]
Like I said in the start of this discussion, I use version 2.04 of the dsmr-parser. I think it is the latest version.

**Comment by Henkkas52** [comment]
@reneklootwijk I found out what is the issue here. I started all kind of error testing, and finally found when a bad reading was detected that the cause of the problem was a corrupt telegram from the smart meter. I tested on failing power L1, L2 or L3 and failing voltage from L1, L2 and L3. Sometimes the attribute was not in the telegram at all, but most of the time it was corrupted in some way, like this example,

![afbeelding](https://github.com/reneklootwijk/node-dsmr/assets/80907008/bff36b01-270b-444f-afbc-708d73aad557)

In this example the error was at power L2, but the corruption appears in many forms, I can't believe this!
So my conclusion is that your parser is working correctly, the cause is corrupt telegrams that sometimes appear.

**Comment by reneklootwijk** [comment]
Thanks for reporting!

Regular corruption is almost always caused by a faulty cable. Some smart meters require a very specific cable. When my DSMR3 meter was replaced by a DSMR5 meter it took 4 different cables before I found the right one.

This error due to corruption also means you disabled crc checking (disableCrcChecking: true) while instantiating the Smartmeter object, otherwise the telegram would have been rejected and not processed at all.

I corrected the parser and now it will display, when using the debug log level, it found a property that is not recognized. A corrupted property id will be logged.
